test-cases/test_bit.cpp

```diff
@@ -1,256 +1,106 @@
-//            0xFD, 0xCB, 0x00, 0x46, 0xFD, 0xCB, 0x00, 0x4E,
-//            0xFD, 0xCB, 0x00, 0x56, 0xFD, 0xCB, 0x00, 0x5E,
-//            0xFD, 0xCB, 0x00, 0x66, 0xFD, 0xCB, 0x00, 0x6E,
-//            0xFD, 0xCB, 0x00, 0x76, 0xFD, 0xCB, 0x00, 0x7E
-//        };
-//
-//        bus.rom_reset();
-//
-//        for (int j = 0; j < memory.size(); j++) {
-//            bus.rom_write(j,memory[j]);
-//        }
-//
-//        if(load_value[i] == 0x00){
-//            for(uint16_t byte_counter = 0; byte_counter < expected_flag_values_00.size(); byte_counter++){
-//                bus.cpu.instruction_cycle();
-//
-//                if(bus.cpu.flag_register != expected_flag_values_00[byte_counter]){
-//                    compare_flag_values = true;
-//                    std::cout << "VALUE: " << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(load_value[i]) << '\t';
-//
-//                    std::cout << "OPCODE: " << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(bus.cpu.opcode) << '\t';
-//
-//                    std::cout << "Flag: 0x" << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(bus.cpu.flag_register) << '\n';
-//                }
-//                ASSERT_EQ(compare_flag_values, false);
-//            }
-//        }
-//        else if(load_value[i] == 0xFF){
-//            for(uint16_t byte_counter = 0; byte_counter < expected_flag_values_FF.size(); byte_counter++){
-//                bus.cpu.instruction_cycle();
-//
-//                if(bus.cpu.flag_register != expected_flag_values_FF[byte_counter]){
-//                    compare_flag_values = true;
-//                    std::cout << "VALUE: " << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(load_value[i]) << '\t';
-//                    std::cout << "OPCODE: " << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(bus.cpu.opcode) << '\t';
-//
-//                    std::cout << "Flag: 0x" << std::setfill('0') << std::setw(2) << std::hex << std::uppercase
-//                              << static_cast<int>(bus.cpu.flag_register) << '\n';
-//                }
-//                ASSERT_EQ(compare_flag_values, false);
-//            }
-//        }
-//    }
-//}
\ No newline at end of file
+#include "test_main.hpp"
+
+
+TEST_F(BitTest, BIT_implied_test){
+    std::vector<uint8_t> expected_flag_values_00 = {0x00, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54, 0x54};
+    std::vector<uint8_t> expected_flag_values_FF = {0x00, 0x38, 0x38, 0x38, 0x38, 0x38, 0x38, 0x38, 0xB8};
+    std::vector<uint8_t> load_value = {0x00, 0xFF};
+
+    std::vector<std::pair<uint8_t, std::vector<uint8_t>>> register_instructions = {
```

Comment:
unordered map is probably better here.

Reply:
an interesting idea, however i wouldnt be able to make the unordered map a constexpr. like the other ill convert it to a constexpr std::array instead.